src/renderers/shared/reconciler/instantiateReactComponent.js

```diff
@@ -83,6 +106,8 @@ function instantiateReactComponent(node, parentCompositeType) {
       // represenations. I.e. ART. Once those are updated to use the string
       // representation, we can drop this code path.
       instance = new element.type(element);
+    } else if (isStatelessComponentType(element.type)) {
```

Comment:
This won't work since we allow module pattern components that may not have a visible prototype. E.g:

```js
function Foo() {
  return {
    render() {
      return <div />;
    }
  };
}
```

I'm surprised this wasn't covered by a unit test.